crates/forge/tests/e2e/workspaces.rs

```diff
@@ -653,3 +663,236 @@ fn virtual_workspace_missing_package() {
 
     assert!(stdout.contains("Failed to find any packages matching the specified filter"));
 }
+
+#[test]
+fn root_workspace_for_entire_workspace_with_filter() {
+    let temp = setup_hello_workspace();
+
+    let output = test_runner(&temp)
+        .args(["--workspace", "simple"])
+        .assert()
+        .success();
+
+    assert_stdout_contains(
+        output,
+        indoc! {r"
+        [..]Compiling[..]
+        [..]Compiling[..]
+        [..]Compiling[..]
+        [..]Compiling[..]
+        [..]Compiling[..]
+        [..]Finished[..]
+        
+        
+        Collected 1 test(s) from addition package
+        Running 0 test(s) from src/
+        Running 1 test(s) from tests/
+        [PASS] addition_integrationtest::nested::simple_case [..]
+        Tests: 1 passed, 0 failed, 0 ignored, 4 filtered out
+        
+        
+        Collected 0 test(s) from fibonacci package
+        Running 0 test(s) from src/
+        Running 0 test(s) from tests/
+        Tests: 0 passed, 0 failed, 0 ignored, 6 filtered out
+        
+        
+        Collected 1 test(s) from hello_workspaces package
+        Running 1 test(s) from src/
+        [PASS] hello_workspaces::tests::test_simple [..]
+        Running 0 test(s) from tests/
+        Tests: 1 passed, 0 failed, 0 ignored, 2 filtered out
+        
+        
+        Tests summary: 2 passed, 0 failed, 0 ignored, 12 filtered out
+        "},
+    );
+}
+
+#[test]
+fn virtual_workspace_for_entire_workspace_with_filter() {
+    let temp = setup_virtual_workspace();
+
+    let output = test_runner(&temp)
+        .args(["--workspace", "simple"])
```

Comment:
For virtual workspaces you shouldn't need the `--workspace` flag and it should still run all packages.